src/lib.rs

```diff
@@ -1,73 +1,86 @@
 extern crate glob;
 extern crate notify;
 extern crate notify_rust;
 extern crate regex;
 #[macro_use]
-extern crate error_chain;
+extern crate structopt;
 
-use clap::{App, Arg, SubCommand};
-
-mod config;
-mod errors;
 mod reactor;
 mod report;
 mod report_builder;
-use config::ConfigBuilder;
-use reactor::Reactor;
 
-pub fn run() {
-    let matches = App::new("cargo")
-        .bin_name("cargo")
-        .help_message("")
-        .version_message("")
-        .subcommand(
-            SubCommand::with_name("testify")
-            .version("0.2.0")
-            .author("Sergey Potapov <blake131313@gmail.com>")
-            .about("Automatically runs tests for Rust project and notifies about the result.\nSource code: https://github.com/greyblake/cargo-testify")
-            .arg(Arg::with_name("includes")
-                 .short("i")
-                 .long("include")
-                 .takes_value(true)
-                 .help("Comma separated list of include pattern in addition to the predefined default patterns"))
-            .arg(Arg::with_name("cargo_test_args")
-                 .multiple(true)
-                 .last(true))
-        )
-        .get_matches();
+use reactor::{Reactor, Config as ReactorConfig};
+use glob::Pattern;
+use structopt::StructOpt;
+use std::time::Duration;
+use std::path::PathBuf;
 
-    let cargo_test_args = if let Some(matches) = matches.subcommand_matches("testify") {
-        matches
-            .values_of("cargo_test_args")
-            .map(|vals| vals.collect::<Vec<_>>())
-            .unwrap_or(vec![])
-    } else {
-        vec![]
-    };
+#[derive(StructOpt)]
+#[structopt(name = "cargo", bin_name = "cargo")]
+enum CargoOpt {
+    #[structopt(name = "testify")]
+    Testify(Args)
+}
 
-    let include_patterns = matches
-        .subcommand_matches("testify")
-        .and_then(|m| m.value_of("includes"))
-        .map(|vals| vals.split(',').collect::<Vec<_>>())
-        .unwrap_or(vec![]);
+#[derive(StructOpt)]
+#[structopt(name = "testify")]
+struct Args {
+    #[structopt(
+        long = "include",
+        short = "i",
+        help = "Only run tests when a file matches the pattern(s)",
+        value_name = "PATTERN",
+        default_value = "src/**/*.rs,tests/**/*.rs,Cargo.toml,Cargo.lock,build.rs",
+        raw(use_delimiter = "true"),
+    )]
+    patterns: Vec<Pattern>,
+
+    #[structopt(
+        long = "delay",
+        short = "d",
+        help = "Wait at least for provided delay (ms) before rerunning tests",
+        value_name = "MILLISECONDS",
+        default_value = "300",
+        parse(try_from_str = "duration_from_str"),
+    )]
+    ignore_duration: Duration,
+
+    #[structopt(
+        help = "Arguments to pass to `cargo test`",
+        value_name = "ARGS",
+    )]
+    cargo_test_args: Vec<String>,
+}
+
+pub fn run() {
+    let CargoOpt::Testify(Args {
+        ignore_duration,
+        patterns,
+        cargo_test_args,
+    }) = CargoOpt::from_args();
```

Comment:
It's was not obvious for me from the first glance, but looks cool :)
So, here we're deconstructing CargoOpt and Args struct to get values for `ignore_duration`, `patterns` and `cargo_test_args`?

Reply:
Yeah, and because the variables are the same name, we can plug them directly into `ReactorConfig`. These features are quite beautiful 👍